### ISSUE TITLE
LASFile.metadata is broken?

### ISSUE THREAD

**Issue by kinverarity1** [kinverarity1/lasio#310]
```python
>>> import lasio.examples
>>> las = lasio.examples.open('sample.las')
>>> las.metadata
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\kinve\code\lasio\lasio\las.py", line 602, in metadata
    s.append(item)
  File "C:\Users\kinve\code\lasio\lasio\las_items.py", line 410, in append
    self.assign_duplicate_suffixes(newitem.useful_mnemonic)
AttributeError: 'str' object has no attribute 'useful_mnemonic'
```

**Comment by dcslagel** [comment]
**Triage notes:**

1. first note
It looks like LASFile.metadata is sending LASFile.section names to SectionItems.append().  But SectionItems.append() is currently expecting dictionary with a useful_mnemonic key/value.

Here is the code from LASFile.metadata()
```python
    def metadata(self):
        s = SectionItems()
        for section in self.sections:
            for item in section:
                s.append(item)
        return s
```

It looks like, at least in some cases, 'section' is a string rather than a dictionary object and subsequently 'item' is a single letter from the string.  

2. Second Node 
A working version of passing items to SectionItems.append() is the parse_header_section() function in lasio/reader.py.   It is doing a quite different set of actions but there may be a few specific concepts that should be implemented in LASFile.metadata() to fix the current break

3. Third Note
I ran the sequence of steps that Kent described in the initial error report on various tags 0.8 , v0.12, v0.14, v0.16 and v0.25.  This issue exists on all those tags,  this issue probably always existed and probably not a regression.

DC

**Comment by JustinGOSSES** [comment]
Found this issue when trying to work on notebook using lasio, wellio, and wellioviz together https://github.com/JustinGOSSES/wellio.js/pull/48

When working with multiple logs, wellioviz uses well name to organize them. Lasio's generated JSON drops well header information such as well name.

**Comment by kinverarity1** [comment]
Thanks @justinGOSSES I'll have a look.
________________________________

**Comment by kinverarity1** [comment]
I'll continue the discussion at wellio: https://github.com/JustinGOSSES/wellio.js/pull/48#issuecomment-612327542

**Comment by kinverarity1** [comment]
Fix was released in v0.25.1
